src/descriptor/mod.rs

```diff
@@ -607,39 +479,39 @@ impl<Pk: MiniscriptKey> DescriptorTrait<Pk> for Descriptor<Pk> {
         }
     }
 
+    /// Attempts to produce a non-malleable satisfying witness and scriptSig to spend an
+    /// output controlled by the given descriptor; add the data to a given
+    /// `TxIn` output.
+    pub fn satisfy<S>(&self, txin: &mut TxIn, satisfier: S) -> Result<(), Error>
+    where
+        S: Satisfier<Pk>,
+    {
+        // easy default implementation
+        let (witness, script_sig) = self.get_satisfaction(satisfier)?;
+        txin.witness = Witness::from_vec(witness);
+        txin.script_sig = script_sig;
+        Ok(())
+    }
+
     /// Computes an upper bound on the weight of a satisfying witness to the
-    /// transaction. Assumes all signatures are 73 bytes, including push opcode
-    /// and sighash suffix. Includes the weight of the VarInts encoding the
+    /// transaction.
+    ///
+    /// Assumes all ec-signatures are 73 bytes, including push opcode and
+    /// sighash suffix. Includes the weight of the VarInts encoding the
     /// scriptSig and witness stack length.
-    fn max_satisfaction_weight(&self) -> Result<usize, Error> {
+    ///
+    /// # Errors
+    /// When the descriptor is impossible to safisfy (ex: sh(OP_FALSE)).
+    pub fn max_satisfaction_weight(&self) -> Result<usize, Error> {
         match *self {
             Descriptor::Bare(ref bare) => bare.max_satisfaction_weight(),
-            Descriptor::Pkh(ref pkh) => pkh.max_satisfaction_weight(),
-            Descriptor::Wpkh(ref wpkh) => wpkh.max_satisfaction_weight(),
+            Descriptor::Pkh(ref pkh) => Ok(pkh.max_satisfaction_weight()),
+            Descriptor::Wpkh(ref wpkh) => Ok(wpkh.max_satisfaction_weight()),
             Descriptor::Wsh(ref wsh) => wsh.max_satisfaction_weight(),
             Descriptor::Sh(ref sh) => sh.max_satisfaction_weight(),
             Descriptor::Tr(ref tr) => tr.max_satisfaction_weight(),
         }
```

Comment:
Style-nit but I prefer to write this as:

```suggestion
        let weight = match *self {
            Descriptor::Bare(ref bare) => bare.max_satisfaction_weight(),
            Descriptor::Pkh(ref pkh) => pkh.max_satisfaction_weight()?,
            Descriptor::Wpkh(ref wpkh) => wpkh.max_satisfaction_weight()?,
            Descriptor::Wsh(ref wsh) => wsh.max_satisfaction_weight(),
            Descriptor::Sh(ref sh) => sh.max_satisfaction_weight(),
            Descriptor::Tr(ref tr) => tr.max_satisfaction_weight(),
        };

	Ok(weight)
```

Reply:
Yes that is nicer, thanks.